### PR TITLE
Store post draft in redux, add post topic selector

### PR DESCRIPTION
**Pull request by calster00**
**Issue link**

[#156 Article Topics](https://github.com/ita-social-projects/dokazovi-fe/issues/156)
[#203 Save new post draft to redux](https://github.com/ita-social-projects/dokazovi-fe/issues/203)

**Story link**

[#122 Story](https://github.com/ita-social-projects/dokazovi-be/issues/122)

## Summary of issue

- [x]  Add a menu for topic selection for new posts
- [x]  Save post draft to redux

## Summary of change

- [x] create redux state slice for ArticleCreationView & NoteCreationView
- [x] save the following state to redux (on change w/ debounce):
    * post topics
    * post title
    * post body - editor content (html string)
    * post preview

### CODE REVIEW

File: src/modules/postCreation/store/postCreationSlice.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  topics: ICheckboxes;
+  title?: string;
+  htmlContent: string;
+  preview: string;
+}
+
+export interface IPostCreationState {
+  [PostTypeEnum.ARTICLE]: INewPostDraft;
+  [PostTypeEnum.DOPYS]: INewPostDraft;
+}
```
Comment (SergeOlabin) [inline src/modules/postCreation/store/postCreationSlice.ts:16]:
```suggestion
export interface IPostCreationState {
  [type in PostTypeEnum]: INewPostDraft;
}
```

Comment (SergeOlabin) [reply src/modules/postCreation/store/postCreationSlice.ts:16]: try this

File: src/modules/postCreation/store/postCreationSlice.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        value: INewPostDraft['preview'];
+      }>,
+    ) => {
+      state[action.payload.postType].preview = action.payload.value;
+    },
+  },
+});
+
+export const {
+  // saveNewPostDraft,
```
Comment (SergeOlabin) [inline src/modules/postCreation/store/postCreationSlice.ts:79]: remove

File: src/modules/postCreation/store/postCreationSlice.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  },
+};
+
+export const postCreationSlice = createSlice({
+  name: 'postCreation',
+  initialState,
+  reducers: {
+    // saveNewPostDraft: (state, action: PayloadAction<IPostCreationState>) => {
+    //   state = action.payload;
+    // },
```
Comment (SergeOlabin) [inline src/modules/postCreation/store/postCreationSlice.ts:38]: remove if not needed

File: src/lib/components/FilterForm.tsx [hunk 4]
```diff
@@ -66,19 +73,21 @@ export const FilterForm: React.FC<IFilterFormProps> = (props) => {
...
-                      id="All"
-                      checked={allChecked}
-                      onChange={onCheckboxAllChange}
-                      name="All"
-                    />
-                  }
-                  label="Всі"
-                  key="All"
-                />
+                {allChecked !== undefined && (
```
Comment (SergeOlabin) [inline src/lib/components/FilterForm.tsx:76]: you cna use just 
```suggestion
                {!!allChecked && (
```

**Review by SergeOlabin — APPROVED** [review]

File: src/modules/postCreation/store/postCreationSlice.ts (hunk 3 above)
Comment (calster00) [reply src/modules/postCreation/store/postCreationSlice.ts:38]: left this for comparison purpose, i'll remove commented sections before merge

File: src/lib/components/FilterForm.tsx (hunk 4 above)
Comment (calster00) [reply src/lib/components/FilterForm.tsx:76]: here i need to distinguish between `false` and `undefined`  – when it's toggled to false vs when this prop is not passed by parent component

File: src/modules/postCreation/store/postCreationSlice.ts (hunk 1 above)
Comment (calster00) [reply src/modules/postCreation/store/postCreationSlice.ts:16]: that wouldn't work for some unclear reason
Снимок экрана 2021-01-18 в 13 52 47

**Review by DelFlumen — APPROVED** [review]

**Pull request merged**